src/main/java/attendance/model/attendance/log/AttendanceLogs.java

```diff
@@ -0,0 +1,151 @@
+
+public record AttendanceLogs(
+        List<AttendanceLog> values
+) {
+
+    public AttendanceLogs(final List<AttendanceLog> values) {
+        this.values = new ArrayList<>(values);
+    }
+
+    public static AttendanceLogs fromLocalDateTimes(
+            final List<LocalDateTime> dateTimes,
+            final CampusOperationPolicy campusOperationPolicy
+    ) {
+
+        final List<AttendanceLog> attendanceLogs = dateTimes.stream()
+                .map(dateTime -> AttendanceLog.fromDateTime(dateTime, campusOperationPolicy))
+                .toList();
+
+        return new AttendanceLogs(attendanceLogs);
+    }
+
+    public AttendanceLogs getAllAttendanceLogsFromTo(
+            final LocalDate from,
+            final LocalDate to,
+            final CampusOperationPolicy campusOperationPolicy
+    ) {
+
+        return new AttendanceLogs(Stream.concat(
+                        values.stream(),
+                        getAbsenceAttendanceLogs(from, to, campusOperationPolicy).stream()
+                )
+                .sorted(Comparator.comparing(AttendanceLog::getDate))
+                .toList());
+    }
+
+    private List<AttendanceLog> getAbsenceAttendanceLogs(
+            final LocalDate from,
+            final LocalDate to,
+            final CampusOperationPolicy campusOperationPolicy
+    ) {
+
+        return from.datesUntil(to)
+                .filter(campusOperationPolicy::isOpenDate)
+                .filter(date -> !containsDate(date))
+                .map(date -> AttendanceLog.fromAbsenceDate(date, campusOperationPolicy))
+                .toList();
+    }
+
+    private boolean containsDate(final LocalDate date) {
+        return values.stream().anyMatch(attendanceLog -> attendanceLog.isSameDate(date));
+    }
+
+    public Map<AttendanceStatus, Integer> calculateAttendanceStatusStatistics() {
+        return AttendanceStatus.getStatistics(getAllAttendanceStatuses());
+    }
+
+    public List<AttendanceStatus> getAllAttendanceStatuses() {
+        return values.stream()
+                .map(AttendanceLog::getAttendanceStatus)
+                .toList();
+    }
+
+    public void add(final AttendanceLog attendanceLog) {
+        values.add(attendanceLog);
+    }
+
+    public int getPolicyAppliedAbsenceCount() {
+        return getAbsenceCount() + getLateCount() / 3;
+    }
+
+    public int getAbsenceCount() {
+        return Math.toIntExact(
+                values.stream()
+                        .filter(this::isAbsence)
+                        .count()
+        );
+    }
+
+    public int getPolicyAppliedLateCount() {
+        return getLateCount() % 3;
+    }
+
+    public int getLateCount() {
+        return Math.toIntExact(
+                values.stream()
+                        .filter(this::isLate)
+                        .count()
+        );
+    }
+
+    private boolean isLate(AttendanceLog attendanceLog) {
+        return attendanceLog.getAttendanceStatus() == AttendanceStatus.LATE;
```

Comment:
```suggestion
        return attendanceLog.isLate();
```
AttendanceLog에 AttendanceStatus 비교 메서드를 만들면 AttendanceLogs에 isLate 메서드를 만들지 않아도 stream에서 처리해볼 수 있을 것 같아요. 😄 
아래 isAbsence 메서드도 마찬가지입니다!

Reply:
감사합니다 ㅎㅎ 설계 중에 놓친 부분이 아직 꽤 있네요...

Reply:
저도 맨날 놓치고 나중에 리팩토링 한답니다 ㅋㅋ.. 👍